MHVLibTests/Connection/MHVConnectionTests.m

```diff
@@ -0,0 +1,264 @@
+#import "MHVSodaConnection.h"
+#import "MHVRestRequest.h"
+#import "MHVMethod.h"
+#import "MHVHttpServiceProtocol.h"
+#import "MHVConfiguration.h"
+#import "MHVClientFactory.h"
+#import "MHVKeychainServiceProtocol.h"
+#import "MHVShellAuthServiceProtocol.h"
+#import "MHVPlatformClientProtocol.h"
+#import "MHVSessionCredentialClientProtocol.h"
+#import "MHVPersonClientProtocol.h"
+#import "MHVApplicationCreationInfo.h"
+#import "MHVSessionCredential.h"
+#import "MHVPersonInfo.h"
+#import "Kiwi.h"
+
+static NSString *const kDefaultSharedSecret = @"TESTSHAREDSECRET";
+static NSString *const kDefaultToken = @"TESTTOKEN";
+
+@interface MHVConnection (Testing)
+
+@property (nonatomic, strong, nullable) MHVSessionCredential *sessionCredential;
+
+@end
+
+SPEC_BEGIN(MHVConnectionTests)
+
+describe(@"MHVConnectionTests", ^
+{
+    // Mocks
+    KWMock<MHVHttpServiceProtocol> *httpService = [KWMock mockForProtocol:@protocol(MHVHttpServiceProtocol)];
+    id<MHVKeychainServiceProtocol> keychainService = [KWMock mockForProtocol:@protocol(MHVKeychainServiceProtocol)];
+    id<MHVShellAuthServiceProtocol> authService = [KWMock mockForProtocol:@protocol(MHVShellAuthServiceProtocol)];
+    
+    MHVClientFactory *clientFactory = [MHVClientFactory mock];
+    id<MHVPlatformClientProtocol> platformClient = [KWMock mockForProtocol:@protocol(MHVPlatformClientProtocol)];
+    id<MHVSessionCredentialClientProtocol> credentialClient = [KWMock mockForProtocol:@protocol(MHVSessionCredentialClientProtocol)];
+    id<MHVPersonClientProtocol> personClient = [KWMock mockForProtocol:@protocol(MHVPersonClientProtocol)];
+    [(id)clientFactory stub:@selector(platformClientWithConnection:) andReturn:platformClient];
+    [(id)clientFactory stub:@selector(credentialClientWithConnection:) andReturn:credentialClient];
+    [(id)clientFactory stub:@selector(personClientWithConnection:) andReturn:personClient];
+    
+    // Test Connection
+    MHVConnection *testConnection = [[MHVSodaConnection alloc] initWithConfiguration:[MHVConfiguration new]
+                                                                       clientFactory:clientFactory
+                                                                         httpService:httpService
+                                                                     keychainService:keychainService
+                                                                    shellAuthService:authService];
+    
+    // Set service and credential for tests
+    testConnection.serviceInstance = [[MHVInstance alloc] init];
+    testConnection.serviceInstance.healthServiceUrl = [NSURL URLWithString:@"https://test.url"];
+    
+    testConnection.sessionCredential = [[MHVSessionCredential alloc] initWithToken:kDefaultToken sharedSecret:kDefaultSharedSecret];
+    
+    // Requested values
+    __block NSURL *requestedURL;
+    __block NSString *httpMethod;
+    __block NSDictionary *requestedHeaders;
+    __block NSData *requestedBody;
```

Comment:
I think you are going to need to set these values to a default state beforeEach() - Otherwise you may end up with false positives/negatives from a leftover state.